src/hotspot/share/logging/logFileOutput.cpp

```diff
@@ -378,81 +379,81 @@ void LogFileOutput::rotate() {
 char* LogFileOutput::make_file_name(const char* file_name,
                                     const char* pid_string,
                                     const char* timestamp_string) {
+  char hostname_string[HostnameBufferSize];
   char* result = nullptr;
 
-  // Lets start finding out if we have any %d and/or %t in the name.
+  // Lets start finding out if we have any %p, %t and/or %hn in the name.
   // We will only replace the first occurrence of any placeholder
   const char* pid = strstr(file_name, PidFilenamePlaceholder);
   const char* timestamp = strstr(file_name, TimestampFilenamePlaceholder);
+  const char* hostname = strstr(file_name, HostnameFilenamePlaceholder);
 
-  if (pid == nullptr && timestamp == nullptr) {
+  if (pid == nullptr && timestamp == nullptr && hostname == nullptr) {
     // We found no place-holders, return the simple filename
     return os::strdup_check_oom(file_name, mtLogging);
   }
 
   // At least one of the place-holders were found in the file_name
-  const char* first = "";
-  size_t first_pos = SIZE_MAX;
-  size_t first_replace_len = 0;
-
-  const char* second = "";
-  size_t second_pos = SIZE_MAX;
-  size_t second_replace_len = 0;
-
-  // If we found a %p, then setup our variables accordingly
+  size_t result_len =  strlen(file_name);
   if (pid != nullptr) {
-    if (timestamp == nullptr || pid < timestamp) {
-      first = pid_string;
-      first_pos = pid - file_name;
-      first_replace_len = strlen(PidFilenamePlaceholder);
-    } else {
-      second = pid_string;
-      second_pos = pid - file_name;
-      second_replace_len = strlen(PidFilenamePlaceholder);
-    }
+    result_len -= strlen(PidFilenamePlaceholder);
+    result_len += strlen(pid_string);
   }
-
   if (timestamp != nullptr) {
-    if (pid == nullptr || timestamp < pid) {
-      first = timestamp_string;
-      first_pos = timestamp - file_name;
-      first_replace_len = strlen(TimestampFilenamePlaceholder);
-    } else {
-      second = timestamp_string;
-      second_pos = timestamp - file_name;
-      second_replace_len = strlen(TimestampFilenamePlaceholder);
+    result_len -= strlen(TimestampFilenamePlaceholder);
+    result_len += strlen(timestamp_string);
+  }
+  if (hostname != nullptr) {
+    if (!os::get_host_name(hostname_string, sizeof(hostname_string))) {
+      int res = jio_snprintf(hostname_string, sizeof(hostname_string), "%s", HostnameFilenamePlaceholder);
```

Comment:
This seems overkill just to write `%hn` into `hostname_string. And on error shouldn't we just replace the `%hn` with e.g. `unknown-host`?

Reply:
fixed